src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -541,15 +584,15 @@ void ObjectSynchronizer::enter(Handle obj, BasicLock* lock, JavaThread* current)
         // be visible <= the ST performed by the CAS.
         lock->set_displaced_header(mark);
         if (mark == obj()->cas_set_mark(markWord::from_pointer(lock), mark)) {
-          return;
+          return true;
         }
         // Fall through to inflate() ...
```

Comment:
This comment and the one at line 579 are no longer clear because we don't inflate in this function.